app/controllers/claims/support/schools_controller.rb

```diff
@@ -30,6 +32,10 @@ def create
 
   private
 
+  def redirect_to_school_options
+    redirect_to new_claims_support_school_option_path(school: { search_param: school_params[:urn] })
+  end
```

Comment:
I wonder if this could be more cleanly implemented by considering the JavaScript version to be a "progressive enhancement".

Rhetorical question: If we were building this entire journey from scratch, without any JavaScript, is this how we'd build/structure it?

I think there could be a way for this interaction to be more simply implemented within a single controller. Something like:

```ruby
class SchoolsController
  def new
    @school_form = SchoolOnboardingForm.new
    @schools = search_results if search_query.present?
  end

  private

  def search_query
    params[:search_query]
  end

  def search_results
    School.search_name_urn_postcode(search_query).map(&:decorate)
  end
end
```

So the view `new.html.erb` receives search results _if a search has been performed_, otherwise it shows the default page.

And then we might not actually need a separate 'school options' controller with a duplicate `check` action and views, etc.

What do you think?

Reply:
The search-options#new controller method comes after the `schools#new` method but before the `schools#check` method. So we can't reuse the `schools#new` method.

We need a new one. I see there's interest in moving this in the `schools controller` so I'll see how I can put it there

Reply:
OK, I've added the methods that are neccessary to the schools controller, we can't reuse the `new` or the `check` methods. 

- The `new` method we can't use because the radio form happens after that.
- The `check` method we can't use because we need to redirect away from it when a param that is only sent by javascript, `params.dig(:school, :search_urn)`. We need to display different error messages in the school form object, so we can't tell the school form object to display this error and also go into the `check` controller method because we are redirecting away when 'javascript is disabled'

Reply:
Catalin and I have just chatted about this. We both think that there could _potentially_ be a refactor – to simplify the JS and non-JS implementations – but we agree that it falls beyond the scope of this PR.

I'm going to mark this as 'resolved', but I may follow up with some more thinking in a future PR.